tedana/decomposition/eigendecomp.py

```diff
@@ -266,16 +266,17 @@ def tedpca(catd, OCcatd, combmode, mask, t2s, t2sG,
 
     if len(ste) == 1 and ste[0] == -1:
         LGR.info('Computing PCA of optimally combined multi-echo data')
-        d = OCcatd[utils.make_min_mask(OCcatd[:, np.newaxis, :])][:, np.newaxis, :]
+        d = OCcatd[mask, :][:, np.newaxis, :]
     elif len(ste) == 1 and ste[0] == 0:
         LGR.info('Computing PCA of spatially concatenated multi-echo data')
-        d = catd[mask].astype('float64')
+        d = catd[mask, ...].astype('float64')
     else:
         LGR.info('Computing PCA of echo #%s' % ','.join([str(ee) for ee in ste]))
-        d = np.stack([catd[mask, ee] for ee in ste - 1], axis=1).astype('float64')
+        d = np.stack([catd[mask, ee, :] for ee in ste - 1], axis=1).astype('float64')
 
     eim = np.squeeze(eimask(d))
     d = np.squeeze(d[eim])
+    d = np.squeeze(d)
```

Comment:
Sorry, why do we need this second `np.squeeze` now ?

Reply:
Whoops. That's a holdover from when I was looking at the eimask as the potential problem. Removed now.